multicluster/controllers/multicluster/core/local_cluster_manager.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+)
+
+/**
+ *  This file contains implementation to manage access to common area in the local cluster backed by a namespace
+ */
+
+type LocalClusterManager interface {
+	// Access to write into the common area of the local cluster
+	// Access to read is likely not needed as the reconciler already is setup to notify changes
+	common.CommonArea
+
+	// Start the local cluster manager and block on a go routine
+	Start() error
+
+	// cleanup any state and Stop the local cluster manager
+	Stop() error
+}
+
+type localClusterManager struct {
+	client.Client
+
+	// this is the local cluster ID
+	clusterID common.ClusterID
+	// namespace for the cluster manager to operate in, same as the
+	// namespace in which the cluster set is defined
+	namespace string
+	// Function to stop this manager
+	stopFunc context.CancelFunc
+}
+
+func NewLocalClusterManager(client client.Client, clusterID common.ClusterID, namespace string) LocalClusterManager {
+	return &localClusterManager{
+		Client:    client,
+		clusterID: clusterID,
+		namespace: namespace,
+	}
+}
+
+/**
+ *  LocalClusterManager implementation
+ */
+
+func (i *localClusterManager) Start() error {
```

Comment:
If nothing needed here, should we just remove this func?

Reply:
Sure, I can remove it.